MaxText/maxtext_utils.py

```diff
@@ -124,3 +126,53 @@ def calculate_tflops_prefill(num_model_parameters, prefill_length, config, log=T
           f'\t\tCausal attention TFLOPs: {causal_attention_tflops} ',
           f'({100 * causal_attention_tflops/total_tflops:.2f})% of Total')
   return total_tflops, learnable_weight_tflops, causal_attention_tflops
+
+
+def calc_not_sharded_params(shard,
+                      expected_per_device_num_param,
+                      key,
+                      pspec,
+                      num_devices_sharded
+                      ):
+  """Returns the number of params that aren't sharded."""
+  new_num_p = np.prod(shard.data.shape)
+
+  if new_num_p == expected_per_device_num_param:
+    print(f'Input is sharded over {num_devices_sharded} devices as expected!')
+  else:
+    print(f'Expected {expected_per_device_num_param} params but got {new_num_p}')
```

Comment:
Let's match style as use max_logging.

Reply:
But as I say above this script should be totally silent unless you assert.

Reply:
I thought of using max_logging, but kept is consistent with "print" statements in the maxtext_utils.py. I will remove printing from here and will switch other prints to max_logging then.